cmd/cli/plugin/secret/README.md

```diff
@@ -171,3 +171,63 @@ the "secret registry" can be used to:
      Updated registry secret 'test-secret' in namespace 'test-ns'
      Unexported registry secret 'test-secret' from all namespaces
    ```
+
+## Workflow for adding a private package repository and installation of a private package
+
+You can add a private package repository and install a private package using the following procedure:
+
+1. First, create the namespace in which the secret is getting added to:
+
+   ```sh
+   kubectl create namespace <NAMESPACE>
+   ```
+
+2. Before adding a private package repository, registry secret should be added to the cluster:
+
+   ```sh
+   tanzu secret registry add <SECRET-NAME> --server <PRIVATE-REGISTRY> --username <USERNAME> --namespace <SECRET-NAMESPACE> --password <PASSWORD> -y
+   ```
+
+3. In case you want to add the private repository in a different namespace than the namespace in which the secret was added to, you need o export the secret to all other namespaces. Please be aware that by doing so, the given secret contents will be available to ALL users in ALL namespaces. Please ensure that included registry credentials allow only read-only access to the registry with minimal necessary scope. You can export the secret to other namespaces by running:
+
+   ```sh
+   tanzu secret registry update <SECRET-NAME> --namespace <SECRET-NAMESPACE> --export-to-all-namespaces=true -y
```

Comment:
It would be nice to have the sample output of commands.When the doc writers try it out, they know they are going on right track by comparing the output